src/stdlib/decode_punycode.rs

```diff
@@ -71,6 +71,10 @@ impl FunctionExpression for DecodePunycodeFn {
         let value = self.value.resolve(ctx)?;
         let string = value.try_bytes_utf8_lossy()?;
 
+        if !string.contains(PUNYCODE_PREFIX) {
```

Comment:
I wonder if we can generally assume this.

Reply:
I had trouble finding this in the punycode spec and then I remembered it is not in punycode spec, but it is used in IDN only, not in punycode in general. I think this might be safe to use, since by default this already validates if the value is a valid domain name (`validate: false` can be passed to skip that).

I could update the docs if it is not clear in the docs that this was the intended behavior.

Reply:
Let's mention it in the docs.